app/src/main/java/com/cornellappdev/uplift/ui/components/reporting/SubmitButton.kt

```diff
@@ -21,16 +24,26 @@ import com.cornellappdev.uplift.util.montserratFamily
  */
 @Composable
 fun SubmitButton(
-    onSubmit: () -> Unit
+    onSubmit: () -> Unit,
+    enabled: Boolean
 ) {
+    val enabledAlpha by animateFloatAsState(
+        targetValue = if (enabled) 1f else 0.5f,
+        label = "enabled alpha"
+    )
     Surface(
         color = PRIMARY_YELLOW,
         shape = RoundedCornerShape(38.dp),
         modifier = Modifier
             .width(106.dp)
-            .height(42.dp),
+            .height(42.dp)
+            .alpha(enabledAlpha),
         shadowElevation = 4.dp,
-        onClick = { onSubmit() }
+        onClick = {
+            if (enabled) {
+                onSubmit()
+            }
+        }
```

Comment:
Oh oops sorry for my first suggestion, there's a parameter to `Surface` called `enabled` that should do this logic for us. So instead of barring `onClick` behind `enabled` and having the enabled alpha you should just be able to give the surface `enabled = enabled`.